assessments/projects/rook/self-assessment.md

```diff
@@ -0,0 +1,374 @@
+
+Rook automates deployment and management of Ceph to provide self-managing, self-scaling, and self-healing storage services. The Rook operator does this by building on Kubernetes resources to deploy, configure, provision, scale, upgrade, and monitor Ceph.
+
+The status of the Ceph storage provider is **Stable**. Features and improvements will be planned for many future versions. Upgrades between versions are provided to ensure backward compatibility between releases.
+
+### Actors
+These are the individual parts of your system that interact to provide the 
+desired functionality.  Actors only need to be separate, if they are isolated
+in some way.  For example, if a service has a database and a front-end API, but
+if a vulnerability in either one would compromise the other, then the distinction
+between the database and front-end is not relevant.
+
+The means by which actors are isolated should also be described, as this is often
+what prevents an attacker from moving laterally after a compromise.
+
+![Rook Components on Kubernetes](Rook%20High-Level%20Architecture.png)
+
+- `Rook Operator`
+	- `Ceph CSI Driver`
+	- `Ceph Daemons`
+
+#### `Rook Operator`
+The Rook operator is a simple container that has all that is needed to bootstrap and monitor the storage cluster. The operator will start and monitor Ceph monitor pods, the Ceph OSD daemons to provide RADOS storage, as well as start and manage other Ceph daemons. The operator manages CRDs for pools, object stores (S3/Swift), and filesystems by initializing the pods and other resources necessary to run the services.
+
+The operator will monitor the storage daemons to ensure the cluster is healthy. Ceph mons will be started or failed over when necessary, and other adjustments are made as the cluster grows or shrinks. The operator will also watch for desired state changes specified in the Ceph custom resources (CRs) and apply the changes.
+
+Rook automatically configures the Ceph-CSI driver to mount the storage to your pods. The rook/ceph image includes all necessary tools to manage the cluster. Rook is not in the Ceph data path. Many of the Ceph concepts like placement groups and crush maps are hidden so you don't have to worry about them. Instead, Rook creates a simplified user experience for admins that is in terms of physical resources, pools, volumes, filesystems, and buckets. Advanced configuration can be applied when needed with the Ceph tools.
+
+#### `Ceph CSI Driver`
+The Ceph-CSI driver provides the provisioning and mounting of volumes
+
+#### `Ceph Daemons`
+The Ceph daemons run the core storage architecture. See the [Glossary](https://github.com/rook/rook/blob/master/Documentation/Getting-Started/glossary.md#ceph) to learn more about each daemon.
+
+### Actions
+These are the steps that a project performs in order to provide some service
```

Comment:
This section is meant to hold the possible actions in this project. 
Please remove any generic content from the template, and refer to this text as a guide to add relevant content.

Reply:
Generic content removed and actor section significantly expanded in this commit: [fe0d055](https://github.com/dhauss/tag-security/commit/fe0d055e96a6fa26851fc13a7d9d5ed33aa98ed4)

Reply:
Expanded here: [5eb312b](https://github.com/dhauss/tag-security/commit/5eb312bff2c7ec72431122574e782f57036c42fd)